### PR TITLE
Add a json editor

### PR DESCRIPTION
**Pull request by calebcauthon**
### Whats New
* Edit the `config` of a step using a json editor

#### See the json editor
image

#### See the event fire
image

### CODE REVIEW

**Comment by darrencauthon** [comment]
screenshot_021019_090729_am

**Comment by darrencauthon** [comment]
@calebcauthon To be more specific, imagine an editor that lets us edit smaller chunks of our handlers, like this "rules" line:

https://github.com/darrencauthon/fasten/blob/b997bc8881ffab973fd0ad44a4ac1fa728a0c338/app/controllers/events_controller.rb#L119

**Comment by darrencauthon** [comment]
@calebcauthon ... or, we could edit the entire "config" block.

**Comment by calebcauthon** [comment]
The editor I have and the editor you showed are functionally the same as far as I can tell.

File: app/assets/javascripts/event-creation-app.js [hunk 1]
```diff
@@ -37,7 +37,15 @@ function buildEventCreationApp(elementId, diagram) {
 
         edgesToRemove.forEach(edge => {
           diagram.edges.remove(edge.id);
+
+          const nextStepIndex = fromNode.step.next_steps.findIndex(thisStep => {
+            return thisStep.id === edge.to;
+          });
+
+          fromNode.step.next_steps.splice(nextStepIndex, 1);
```
Comment (calebcauthon) [inline app/assets/javascripts/event-creation-app.js]: This fixes a bug on the editor, when un-linking steps from each other. Previously, the step remained linked when you 'fired' it.

File: app/models/workflow.rb [hunk 2]
```diff
@@ -110,6 +106,15 @@ def start(event)
 
   private
 
+  def self.copy_event_data_from source_event, target_events
+    target_events.each do |target_event|
+      target_event.data = Hash.new unless target_event.data
```
Comment (calebcauthon) [inline app/models/workflow.rb]: Here's the key change...

If `.data` doesnt exist, we give it one. This was previously blowing up for certain event handler types.

File: app/views/sequence/new.html.erb [hunk 3]
```diff
@@ -1,10 +1,42 @@
...
+      var container = document.getElementById("jsoneditor");
+      var options = {
+        onChange: () => onChange()
+      };
+      var editor = new JSONEditor(container, options);
+
+      // set json
+      var json = {
+        'merge_mode': 'merge'
+      };
```
Comment (calebcauthon) [inline app/views/sequence/new.html.erb]: There's a default in the UI... I want to go further and put in config templates for each type.

File: app/views/sequence/new.html.erb [hunk 4]
```diff
@@ -15,12 +47,24 @@
...
-    <input v-model="network.selected.step.type" />
+    <select v-model="network.selected.step.type">
+      <option value="">Please select</option>
+      <option>EventHandler</option>
+      <option>WebRequest</option>
+      <option>JsonParser</option>
+      <option>Trigger</option>
+      <option>EventFormatter</option>
+      <option>Post</option>
+    </select>
```
Comment (calebcauthon) [inline app/views/sequence/new.html.erb]: A dropdown of our event types... obv we can automate this in some way in the future.

**Pull request merged**